src/microk8scluster.py

```diff
@@ -271,6 +271,9 @@ def _refresh_channel(self, _):
         self.model.unit.status = ActiveStatus()
 
     def _coredns_config(self, event):
+        if "coredns_config" not in self.model.config:
```

Comment:
perhaps this might be better? that way, an empty config string will also not cause issues

```suggestion
        if not self.model.config.get("coredns_config"):
```

Reply:
That way we can also not mess with the defaults, new deployments can take care of setting `coredns_config: ""` to get the new behavior.

Reply:
Thanks for the feedback, pushed the change.